### PR TITLE
File upload

### PR DESCRIPTION
**Pull request by abhay-venkatesh**
Added a file upload task category on the authoring side of the crowdresearch platform.

### CODE REVIEW

File: static/css/_task-template.scss [hunk 1]
```diff
@@ -71,6 +71,18 @@
     }
 }
 
+._file-upload {
+  & .md-button {
+    margin-bottom: 20px;
+  }
+  & #progress {
```
Comment (dmorina) [inline static/css/_task-template.scss]: a "," perhaps, lol

File: static/js/task/controllers/task.controller.js [hunk 2]
```diff
@@ -149,13 +175,7 @@
                     if (status === 1) {
                         $mdToast.showSimple('Could not save task.');
                     } else {
-                        if(data[0].hasOwnProperty('message')){
```
Comment (dmorina) [inline static/js/task/controllers/task.controller.js]: because? we don't like error messages anymore :D (?)

File: static/templates/task/base.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<div class="_main-content" style="width: 100%">
```
Comment (dmorina) [inline static/templates/task/base.html]: this reverts a change that has been made recently, revert this file please

File: static/js/task/controllers/task.controller.js (hunk 2 above)
Comment (abhay-venkatesh) [inline static/js/task/controllers/task.controller.js]: that's weird, don't recall deleting this haha

File: static/css/_task-template.scss (hunk 1 above)
Comment (abhay-venkatesh) [inline static/css/_task-template.scss]: between?

Comment (dmorina) [inline static/css/_task-template.scss]: just a hint, if two css rules are exactly the same then you do: `rule1, rule2 { margin-top:15 px;}`

File: static/templates/task/base.html (hunk 3 above)
Comment (abhay-venkatesh) [inline static/templates/task/base.html]: can I revert the file in the browser interface for github or do I have to make another PR after editing in my text editor?

Comment (dmorina) [inline static/templates/task/base.html]: not in the browser: i guess you wanna keep `
+            <form ng-init="template.upload = task.upload"
`
but you have to merge correctly with dev2

Comment (mbernst) [inline static/templates/task/base.html]: I think you just make the changes on your local branch on your machine, then `git push` and the PR should update automatically. @dmorina correct me if I'm wrong.

File: static/css/_task-template.scss [hunk 4]
```diff
@@ -71,6 +71,15 @@
     }
 }
 
+._file-upload {
+  & .md-button {
+    margin-bottom: 20px;
+  }
+  & #progress, & #done {
```
Comment (abhay-venkatesh) [inline static/css/_task-template.scss]: Is this the right syntax for the & operator?

File: static/templates/task/base.html (hunk 3 above)
Comment (dmorina) [inline static/templates/task/base.html]: @mbernst the issue is something else in this case: when he merged develop2 into his branch he discarded the new changes from remote server and accepted only his local changes

File: static/css/_task-template.scss (hunk 4 above)
Comment (dmorina) [inline static/css/_task-template.scss]: does it work?

Comment (abhay-venkatesh) [inline static/css/_task-template.scss]: yeah.

File: static/js/task/controllers/task.controller.js (hunk 2 above)
Comment (abhay-venkatesh) [inline static/js/task/controllers/task.controller.js]: added this back.
